cmd/api/main.go

```diff
@@ -490,3 +365,215 @@ func configureTelemetry(
 		return nil
 	}, nil
 }
+
+func restoreBackup(databaseURL string, backupURL string) error {
+	restorer, err := restorer.NewBackupRestorer(backupURL, databaseURL)
+	if err != nil {
+		return fmt.Errorf("creating restorer: %s", err)
+	}
+
+	log.Info().Msg("starting backup restore")
+	elapsedTime := time.Now()
+	if err := restorer.Restore(); err != nil {
+		return fmt.Errorf("backup restoration failed: %s", err)
+	}
+	log.Info().Float64("elapsed_time_seconds", time.Since(elapsedTime).Seconds()).Msg("backup restore finished")
+	return nil
+}
+
+func createParser(queryConstraints QueryConstraints) (parsing.SQLValidator, error) {
```

Comment:
An example of sending `QueryConstraints`, and not the full `Config`.
Being focused allows to truly understand what configs or dependencies are minimally needed in different modules.